### PR TITLE
[FIX] account_qr_code_emv, l10n_*: fix emv_qr_vals format

### PR DESCRIPTION
**Pull request by tong-odoo**
Steps to reproduce:

- Create any emv_qr_code. Set merchant name with a Vietname accent
- Set amount to a integer number

Current behaviour:

- An EMV QR Code will be generated with accent merchant name, and the amount include .0 value

Expected behaviour:

- The EMV QR Code change change the accent value to alphabet. And the amount should not contain .0 if the amount is an integer

X-original-commit: 14221a5

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_vn/tests/test_l10n_vn_emv_qr.py [hunk 1]
```diff
@@ -81,4 +81,20 @@ def test_emv_qr_vals(self):
...
+            qr_method=self.emv_qr_invoice.qr_code_method,
+            amount=self.emv_qr_invoice.amount_residual,
+            currency=self.emv_qr_invoice.currency_id,
+            debtor_partner=self.emv_qr_invoice.partner_id,
+            free_communication=unstruct_ref,
+            structured_communication=self.emv_qr_invoice.payment_reference,
+        )
+
+        # Check the whole qr code string
+        self.assertEqual(emv_qr_vals, '00020101021238590010A0000007270129000697042201156607040600001290208QRIBFTTA52040000530370454031005802VN5914aAeEoOiIuUyYđĐ6007Vietnam62170113INV/TEST/000263046744')
```
Comment (vin-odoo) [inline addons/l10n_vn/tests/test_l10n_vn_emv_qr.py:100]: aAeEoOiIuUyYđĐ

I see two d's that should have been escaped but were not

**Review by vin-odoo — CHANGES_REQUESTED** [review]

File: addons/l10n_vn/tests/test_l10n_vn_emv_qr.py (hunk 1 above)
Comment (tong-odoo) [reply addons/l10n_vn/tests/test_l10n_vn_emv_qr.py]: only this two characters are not remove by the library :eyes:

File: addons/l10n_vn/tests/test_l10n_vn_emv_qr.py [hunk 2]
```diff
@@ -81,4 +81,20 @@ def test_emv_qr_vals(self):
         )
 
         # Check the whole qr code string
-        self.assertEqual(emv_qr_vals, '00020101021238590010A0000007270129000697042201156607040600001290208QRIBFTTA5204000053037045405100.05802VN5914company_1_data6007Vietnam62170113INV/TEST/0001630425A9')
+        self.assertEqual(emv_qr_vals, '00020101021238590010A0000007270129000697042201156607040600001290208QRIBFTTA52040000530370454031005802VN5914company_1_data6007Vietnam62170113INV/TEST/00016304E801')
+
+    def test_emv_qr_vals_with_accent_partner(self):
+        self.company_data['company'].partner_id.name = 'áÁéÉóÓíÍúÚýÝđĐ'
```
Comment (vin-odoo) [inline addons/l10n_vn/tests/test_l10n_vn_emv_qr.py:87]: I wonder if we didn't ignore many characters that may not work.

https://altcodeunicode.com/alt-codes-for-vietnamese-letters-with-accents-or-diacritics/

**Review by vin-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**